docs/flatpak-devel.rst

```diff
@@ -0,0 +1,395 @@
+  if profile == 'development'
+    find_program('git')
+    rev_txt = run_command('git','rev-parse','--short','HEAD').stdout().strip()
+    rev = '-@0@'.format(rev_txt)
+    application_id = 'org.example.coolapp.Devel'
+  else
+    rev = ''
+    application_id = 'org.example.coolapp'
+  endif
+
+  config_h = configuration_data()
+  config_h.set_quoted('APPLICATION_ID', application_id)
+  config_h.set_quoted('PROFILE', profile)
+  config_h.set_quoted('GETTEXT_PACKAGE', 'coolapp')
+  config_h.set_quoted('PACKAGE_VERSION', meson.project_version())
+  configure_file(
+     output: 'config.h',
+     configuration: config_h,
+  )
+
+  configuration_inc = include_directories('.')
+
+  subdir('data')
+  subdir('src')
+  subdir('po')
+
+  summary({
+    'Profile': get_option('profile'),
+  }, section: 'Development')
+
+
+The root ``meson-options.txt`` should have something like::
+
+  option('profile', type: 'combo', choices: ['default', 'development'], value: 'default')
+
+Now under ``src/meson.build`` or in any subdirectory this can be
+specified in a target.
+
+.. code-block::
+
+  coolapp = executable('coolapp', 'main.c',
+      include_directories: configuration_inc,
+      dependencies: libcoolapp_dep,
+      install: true,
+  )
+
+Now in ``data/meson.build``, the icon, desktop file and metainfo file
+handling can be specified.
+
+Place a desktop template file called ``org.example.coolapp.desktop.in.in``
+in ``data/`` with the contents::
+
+.. code-block::
+
+  [Desktop Entry]
+  Name=Cool App
+  Exec=coolapp
+  Icon=@icon@
+  Terminal=false
+  Type=Application
+  Categories=Utility;
+  StartupNotify=true
+
+
+The corresponding meson bits for the desktop file should be
+
+.. code-block::
+
+  desktop_conf = configuration_data()
+  desktop_conf.set('icon', application_id)
+  desktop_file = i18n.merge_file(
+    input: configure_file(
+      input: files('org.example.coolapp.desktop.in.in'),
+      output: 'org.example.coolapp.desktop.in',
+      configuration: desktop_conf,
+    ),
+    output: '@0@.desktop'.format(application_id),
+    type: 'desktop',
+    po_dir: '../po',
+    install: true,
+    install_dir: get_option('datadir') / 'applications',
+  )
+
+  desktop_utils = find_program('desktop-file-validate', required: false)
+  if desktop_utils.found()
+    test('Validate desktop file', desktop_utils, args: [desktop_file])
+  endif
+
+Appstream can be handled in the same ``meson.build``. Place a metainfo
+template file in ``data/``. Any tag containing part of the application
+ID should be templated. Ususally for desktop applications this involves
+only the ``id`` and ``launchable`` tag, but the ``icon`` tag, if present
+should also use it.
+
+.. code-block:: yaml
+
+  <?xml version="1.0" encoding="UTF-8"?>
+  <component type="desktop">
+    <id>@appid@</id>
+    ...
+    <launchable type="desktop-id">@appid@.desktop</launchable>
+    ...
+  </component>
+
+The corresponding meson.build part for metainfo should be
+
+.. code-block::
```

Comment:
```suggestion
.. code-block:: meson
```

You could add this to all these code blocks. I'm not sure if our setup has a definition for meson but it is reasonable to assume it could or will some day.

Reply:
It seems to output a bunch of warnings about lexing literal blocks. I avoided adding it. Not sure if that's because of partial meson.build snippets everywhere.

Reply:
Bummer.